### ISSUE TITLE
[Numpy] - Depreciation Warning (np.bool8 is deprecated in numpy 1.24)

### ISSUE THREAD

**Issue by ai-nikolai** [microsoft/TextWorld#323]
When running textworld + gym the following depreciation warning happens: (when typing `help`)
```bash
lib/python3.9/site-packages/gym/utils/passive_env_checker.py:233: DeprecationWarning: `np.bool8` is a deprecated alias for `np.bool_`.  (Deprecated NumPy 1.24)
  if not isinstance(done, (bool, np.bool8)):
```

_Software Versions:_
1. textworld==1.5.4
2. numpy==1.26.2
3. gym==0.25.2
4. gym-notices==0.0.8

**Comment by MarcCote** [comment]
Thanks for pointing this out. I might consider making a new release and bump up the dependencies and fix all those deprecation warnings.

**Comment by MarcCote** [comment]
Dropping `gym` dependency.
Fixed by #341
